src/middleware/cors/service.rs

```diff
@@ -1,41 +1,211 @@
-use super::Config;
+use super::{Config, CorsResource};
 
-use futures::{Poll};
-use http;
+use futures::{Async, Future, Poll};
+use http::{self, HeaderMap, StatusCode};
+use http::{Request, Response};
 use tower_service::Service;
 
 use std::sync::Arc;
 
+#[derive(Debug)]
 pub struct CorsService<S> {
     inner: S,
     config: Arc<Config>,
 }
 
 impl<S> CorsService<S> {
     pub(super) fn new(inner: S, config: Arc<Config>) -> CorsService<S> {
-        CorsService {
-            inner,
-            config,
-        }
+        CorsService { inner, config }
     }
 }
 
-use util::http::HttpService;
-use http::{Request, Response};
-
 impl<S> Service for CorsService<S>
-where S: HttpService,
+where
+    S: ::util::http::HttpService,
 {
     type Request = Request<S::RequestBody>;
-    type Response = Response<S::ResponseBody>;
+    type Response = Response<Option<S::ResponseBody>>;
```

Comment:
I don't think this needs to be changed, since it's quite clear what it means, but just noting this isn't the first place we've needed to be able to make some "empty" body in a generic way...

hyper's `Client::get` requires the body generic to have a `Default` bounds, hoping that a default is "empty". I'm not sure hyper's way is best, just putting down for completeness.